sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import.service.ts

```diff
@@ -0,0 +1,127 @@
+import { Injectable } from "@nestjs/common";
+import {
+  ApplicationBulkWithdrawalHeader,
+  ApplicationWithdrawalTextValidationResult,
+} from "./application-bulk-withdrawal-import-text.models";
+import { DataSource, In, Repository } from "typeorm";
+import { Application } from "@sims/sims-db";
+import {
+  ApplicationBulkWithdrawalImportBusinessValidationModel,
+  ApplicationData,
+} from "./application-bulk-withdrawal-import-business-validation.models";
+import { InjectRepository } from "@nestjs/typeorm";
+
+type ApplicationDataMap = Record<string, ApplicationData>;
+
+/**
+ * Handles the application withdrawal bulk insert preparation.
+ */
+@Injectable()
+export class ApplicationBulkWithdrawalImportService {
+  constructor(
+    private readonly dataSource: DataSource,
+    @InjectRepository(Application)
+    private readonly applicationRepo: Repository<Application>,
+  ) {}
+
+  /**
+   * Generates the application bulk withdrawal business validation model.
+   * @param textValidations text validation models to be converted to the business validation models.
+   * @param textHeader text header model.
+   * @param institutionId institution id.
+   * @param allowedLocationIds set of allowed location ids for the given institution.
+   * @returns application bulk withdrawal business models to be validated and persisted.
+   */
+  async generateApplicationBulkWithdrawalValidationModels(
+    textValidations: ApplicationWithdrawalTextValidationResult[],
+    textHeader: ApplicationBulkWithdrawalHeader,
+    institutionId: number,
+    allowedLocationIds: number[],
+  ): Promise<ApplicationBulkWithdrawalImportBusinessValidationModel[]> {
+    const applicationDataMap: ApplicationDataMap = {};
+    const applicationNumbers = textValidations.map(
+      (textValidation) => textValidation.textModel.applicationNumber,
+    );
+    const applicationData =
+      await this.getApplicationBulkWithdrawalValidationDetails(
+        applicationNumbers,
+        institutionId,
+      );
+    applicationData.forEach((record) => {
+      applicationDataMap[record.applicationNumber] = {
+        applicationStatus: record.applicationStatus,
+        isArchived: record.isArchived,
+        sin: record.student.sinValidation.sin,
+        locationId: record.location.id,
+        locationCode: record.location.institutionCode,
+        hasPreviouslyBeenWithdrawn: !!record.studentScholasticStandings.length,
+      };
+    });
+    const businessValidationModels: ApplicationBulkWithdrawalImportBusinessValidationModel[] =
+      [];
+    textValidations.forEach((textValidation) => {
+      const businessValidationModel =
+        {} as ApplicationBulkWithdrawalImportBusinessValidationModel;
+      const applicationData =
+        applicationDataMap[textValidation.textModel.applicationNumber];
+      if (applicationData) {
+        businessValidationModel.applicationFound = true;
+        businessValidationModel.applicationBelongsToInstitution =
```

Comment:
I didn't get this, what does institution admin have to do with `applicationBelongsToInstitution` here?

Reply:
if the plan is to check if the application belongs to the institution, it is already achieved in the query where clause right? line 122 - 123

Reply:
Here I would need some clarification also. If the query is also checking if the institution, an application not found can be because ii belongs to SFAS or because it does not belong to the institution. Is it right to assume that we do not need to have this distinction?

Reply:
The message in AC, "Record was not found and skipped" me assume that

![image](https://github.com/bcgov/SIMS/assets/54600590/3947838d-3488-478f-bfca-a31e780d5943)

Reply:
Removed this property.